src/parse-date.js

```diff
@@ -3,6 +3,26 @@
 var expirationYear = require('./expiration-year');
 var isArray = require('./lib/is-array');
 
+function getLen(value) {
+  if (value[0] === '0') {
+    return 2;
+  }
+
+  if (value.length > 5) {
+    return 2;
+  }
+
+  if (value[value.length - 1] === '0') {
+    return 1;
+  }
+
+  if (value.length < 4) {
```

Comment:
I don't understand why we are assuming that if a date is 3 characters or less, then the month must have a length of 2. What's the reason?

Reply:
described just above in the comments and in the description of the parser for the dates, let me know if you still have questions, I'll try to answer them

Reply:
I think you are misunderstanding what this function does, because the comments you have do not always match the results. 

If this function returns `2` that tells us that the provided month portion of the date string is comprised of 2 digits. IE, a month from 1-9 that has a leading zero. Or a month of 10-12.

If this function returns `1` then the month portion of the date string must be between 1-9, because there's no way that a month of 10-12 has only one digit.

This case in particular, you have `'122' => {month: '1', year: '22'}`, indicating that the month has a length of 1. But you're returning 2 here, which would mean that `'122'` would instead get parsed as ` {month: '12', year: '2'}`, which is not correct to me.

You even have an explicit test for this: https://github.com/braintree/card-validator/pull/71/files#diff-83a3002eb5d7fed18df4475ccf5fbb21R21

So not sure where the misunderstanding is happening. Is the comment incorrect? Or is your parsing logic?

Reply:
Unfortunately this logic was laid earlier and it covered on how many I understood such cases as '12202' => {month: '1', year: '2202'}